src/Data/Foldable.js

```diff
@@ -0,0 +1,26 @@
+/* global exports */
+"use strict";
+
+// module Data.Foldable
+
+exports.foldrArray = function (f) {
+  return function (init) {
+    return function (xs) {
+      /* jshint maxparams: 2 */
+      return xs.reduceRight(function (acc, x) {
```

Comment:
There's a bunch of ES5 stuff in the libraries now I think. I thought that's what we were doing now, and recommending a shim if <ES5 is needed?

We did actually use a loop for this previously, but I figured this might be better, as in theory these operations can be better optimised by a JS engine than a loop. I guess it doesn't matter if in practice it is slower in every browser/engine though, and we should go back to the loop. I just tried a JSPerf on my OSX/Chrome and the `reduceRight` version is 90% slower.